### PR TITLE
Add flag to enable mutual certificate authentication for old Akka Remote SSL transport

### PR DESCRIPTION
**Pull request by jrudolph**
/cc @wsargent

### CODE REVIEW

**Review by johanandren — APPROVED** [review]

File: akka-remote/src/main/scala/akka/remote/transport/netty/NettySSLSupport.scala [hunk 1]
```diff
@@ -34,25 +37,51 @@ private[akka] class SSLSettings(config: Config) {
...
+      }
+      val rng = createSecureRandom(log)
+
+      val ctx = SSLContext.getInstance(SSLProtocol)
+      ctx.init(keyManagers, trustManagers, rng)
+      ctx
+    } catch {
+      case e: FileNotFoundException    ⇒ throw new RemoteTransportException("Server SSL connection could not be established because key store could not be loaded", e)
+      case e: IOException              ⇒ throw new RemoteTransportException("Server SSL connection could not be established because: " + e.getMessage, e)
+      case e: GeneralSecurityException ⇒ throw new RemoteTransportException("Server SSL connection could not be established because SSL context could not be constructed", e)
```
Comment (patriknw) [inline akka-remote/src/main/scala/akka/remote/transport/netty/NettySSLSupport.scala]: should `Server` still be in these exception messages?

**Review by patriknw — APPROVED** [review]
LGTM

**Comment by ktoso** [comment]
LGTM, can we first merge mine though? https://github.com/akka/akka/pull/21749 ;-)

**Review by ktoso — APPROVED** [review]

File: akka-remote/src/main/scala/akka/remote/transport/netty/NettySSLSupport.scala (hunk 1 above)
Comment (drewhk) [inline akka-remote/src/main/scala/akka/remote/transport/netty/NettySSLSupport.scala]: Good point. I think the message needs to be changed.

**Comment by drewhk** [comment]
Missing test? Shouldn't we have a test that fails if an untrusted system manages to connect?

**Comment by ktoso** [comment]
Can you rebase @jrudolph sorry for that

**Comment by ktoso** [comment]
LGTM :)

**Comment by jrudolph** [comment]
Rebased. And I agree we should add tests before claiming that it actually works. Before I had changed the default to run it with the setting on, so that put more coverage on the new feature but we also need to test the bad case as well. I got stuck on understanding how akka-remote really works while writing the test. When it rejects the connection, you might only see that messages are not transmitted but not why. So, we need 1) test coverage for that feature 2) improve logging no to ignore the exceptions coming out of failed connection attempts.

**Comment by ktoso** [comment]
Please open a ticket for the remaining work 1) and 2) here @jrudolph

**Comment by jrudolph** [comment]
https://github.com/akka/akka/issues/21757
https://github.com/akka/akka/issues/21758

**Pull request merged**